internal/file/file.go

```diff
@@ -13,9 +13,98 @@
 
 package file
 
-import "regexp"
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// ErrNotRegularFile is returned when the file is not an regular file.
+var ErrNotRegularFile = errors.New("not regular file")
+
+// ErrNotDirectory is returned when the path is not a directory.
+var ErrNotDirectory = errors.New("not directory")
 
 // IsValidFileName checks if a file name is cross-platform compatible
 func IsValidFileName(fileName string) bool {
 	return regexp.MustCompile(`^[a-zA-Z0-9_.-]+$`).MatchString(fileName)
 }
+
+// CopyToDir copies the src file to dst. Existing file will be overwritten.
+func CopyToDir(src, dst string) error {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return ErrNotRegularFile
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+
+	if err := os.MkdirAll(dst, 0700); err != nil {
```

Comment:
@shizhMSFT I think we shouldn't trust the original permissions based on security concerns?

Reply:
I understand that certs, configs, keys cannot be read by others. But...how about sharing plugins to others?

For example, assume we want to create a base image of notation with plugins pre-installed. By using `notation plugin install`, those pre-installed plugins will have `root` as the owner and are stored in a shared place. When running this base image, a different user other than `root` will be specified for best practices. Since that user does not have read / exec permission on the plugin, that user cannot invoke the shared plugins.

Reply:
> I understand that certs, configs, keys cannot be read by others. But...how about sharing plugins to others?
> 
> For example, assume we want to create a base image of notation with plugins pre-installed. By using `notation plugin install`, those pre-installed plugins will have `root` as the owner and are stored in a shared place. When running this base image, a different user other than `root` will be specified for best practices. Since that user does not have read / exec permission on the plugin, that user cannot invoke the shared plugins.

@shizhMSFT I see, this is indeed a very interesting scenario. Updating the code.

Reply:
How about making sure its not modifiable apart from user that copied it? If plugin executable is modifiable by any user and system relies on this plugin to sign/verify, any user can modify a plugin to control sign/verify process.